lib/ObjectSerializer.php

```diff
@@ -441,7 +441,7 @@ public static function deserialize($data, $class, $httpHeaders = null)
         }
 
         /** @psalm-suppress ParadoxicalCondition */
-        if (in_array($class, ['\DateTime', 'array', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)) {
+        if (in_array($class, ['\DateTime', '\SplFileObject', 'array', 'bool', 'boolean', 'byte', 'double', 'float', 'int', 'integer', 'mixed', 'number', 'object', 'string', 'void'], true)) {
```

Comment:
@esquivalient in addition to general review, please take a look at this in particular. I don't remember exactly why we removed this from the array previously. As of the last commit we're no longer creating SplFileObject so I'm guessing it won't be a big deal to stop patching this away, but if you remember more details then maybe we'll want to bring that patch back.

Reply:
We talked in-person and we both remembered the related error being very obvious. Since the tests now pass we're not worried about adding this back.